app/models/anonymous_user.rb

```diff
@@ -44,4 +44,8 @@ def email_addresses
   def confirmed_at
     Time.zone.now
   end
+
+  def locked_out?
+    second_factor_locked_at.present? && !lockout_period_expired?
+  end
```

Comment:
This new method on AnonymousUser caught my eye. I didn't catch references to that class elsewhere in the diff. Why does this one method need to be added?

Reply:
Previously you could "decorate" the AnonymousUser instance, after which the decorator would have the `locked_out?` method available. This was used in a test or two. So I needed to add that method back in so that those tests passed.

Reply:
should this just be hardcoded to `false` for anonymous users? they have no second factor

Reply:
```suggestion

  def locked_out?
    false
  end
```